tests/v2/validation/rbac/rbac_test.go

```diff
@@ -59,6 +60,19 @@ func (rb *RBTestSuite) SetupSuite() {
 
 }
 
+func checkRBACResourceExistsForUser(client *rancher.Client, collection *v1.SteveCollection, userName string) error {
+	userId, err := users.GetUserIDByName(client, userName)
+	if err != nil {
+		return err
+	}
+	for _, item := range collection.Data {
+		if item.JSONResp["userName"] == userId {
+			return nil
+		}
+	}
+	return errors.New(collection.Data[0].JSONResp["kind"].(string) + " does not exist for user " + userName + "!")
```

Comment:
This seems like a possible flaky scenario. Why index from zero to get the kind? I think we can make this a little bit more efficient in you list call you should use the list query instead iterating through the whole slice: https://github.com/rancher/rancher/blob/5e1ee26116162ab77998f11e0d53d19663f304c7/tests/v2/integration/steveapi/steve_api_test.go#L2414-L2415

Reply:
Good point. This reminded me, we have a similar querying done by Michael to verify the roll out. we can probably use this function and then request for Project or Cluster Role template bindings, code gets the list of RoleBindings. https://github.com/rancher/rancher/blob/release/v2.8/tests/framework/extensions/users/users.go#L289

Reply:
Yes very similar. Do you mean use the function directly? So export it?

Reply:
this is a very good point. @anupama2501 ?

Reply:
Went through the code and it solves the problem of checking the CRTB/PRTB when a user is created/deleted. So, yes we can mark the ticket as resolved for CRTB/PRTB but for grbs we still need the changes.

Reply:
Removed CRTB/PRTB related changes. Modified the code to list GRB for a user by using list query as suggested by @igomez06